waveform_modes.py

```diff
@@ -315,6 +315,29 @@ def indices(self, ell_m):
             raise ValueError("Input `ell_m` should be an Nx2 sequence of integers")
         return ell_m[:, 0] * (ell_m[:, 0] + 1) - self.ell_min ** 2 + ell_m[:, 1]
 
+
+    def inner_prod(self, b):
```

Comment:
For clarity, would it be better to have a different name for the inner product here and the one in `mode_calculations.py`?

Reply:
I'd be in favor of actually calling this `inner_product`; the missing letters make me worry something bad happened to Leo just as he was typing the function name.  Or maybe even `__matmul__`!  :)

Reply:
Something terrible happened to me—I felt like a fortran programmer and started abbreviating things instead of being verbose!

Not sure I understand the suggestion/reference to __matmul__. The timesteps must also be taken into account as weights for the sum. Can you clarify?

Reply:
> I felt like a fortran programmer

Eh, that's still better than Mark, who programs everything as if it were Perl.

If you `def __matmul__(self, b)`, you'd be able to call it as `a @ b`.  The function would still have access to all of the information in `a` and `b` (including the timesteps), but would just make you feel cooler when you call it.

Of course, since so much of my "documentation" relies on being able to find functions by way of tab completion, it's probably better to stick with an actual name...

Reply:
I did not know about the @ magic or any other magic operators. I think of waveforms as vectors, not matrices, so I don't like the semantics of using @ for this purpose. However, support the idea of overloading addition/subtraction/multiplication by scalar (real or complex) for waveforms.

Reply:
> I think of waveforms as vectors, not matrices

Vectors that *aren't* matrices?!  The `@` operator can handle both, of course.  Anyway, for reasons explained above, I'm happy with this version.

As for overloading scalar operators, I implemented them in GWFrames, but eventually came to the conclusion that it's not actually useful because the action to be taken is ambiguous, and can either already be done easily and with less ambiguity (e.g., `h.data *= 3.14`), or should be given its own function (e.g., rescaling mass or distance).